packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -223,6 +223,10 @@ class NotificationDialog extends React.Component {
       message,
       ...customProps
     } = this.props;
+    if (acceptAction === undefined && primaryAction === undefined && rejectAction === undefined && secondaryAction === undefined && process.env.NODE_ENV !== 'production') {
```

Comment:
You can add this `process.env.NODE_ENV !== 'production'  as the first condition. as this should satisfy first to validate the rest.

Reply:
Changed in 9e2431a